### PR TITLE
Added test for raft_client

### PR DESCRIPTION
**Pull request by zyh-hust**
## What have you changed? (mandatory)
Add a test code

## What are the type of the changes? (mandatory)
Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)
you can run test code with this command:
cargo test test_batch_raft_server_drop

## Does this PR affect documentation (docs) or release note? (mandatory)
no

### CODE REVIEW

File: tests/integrations/server/raft_client.rs [hunk 1]
```diff
@@ -89,3 +89,124 @@ fn test_batch_raft_fallback() {
     }
     pool.shutdown().wait().unwrap();
 }
+
+
+#[test]
+fn test_batch_raft_server_drop() {
```
Comment (hicqu) [inline tests/integrations/server/raft_client.rs]: please rename to `test_raft_client_reconnect`, and comment the function.

File: tests/integrations/server/raft_client.rs [hunk 2]
```diff
@@ -89,3 +89,211 @@ fn test_batch_raft_fallback() {
...
+    #[derive(Clone)]
+    struct MockKvForRaft(Arc<AtomicUsize>);
+
+    impl MockKvService for MockKvForRaft {
+        fn raft(
```
Comment (hicqu) [inline tests/integrations/server/raft_client.rs]: I think we don't need to implement this, just put logic into `batch_raft` is fine.

File: tests/integrations/server/raft_client.rs [hunk 3]
```diff
@@ -89,3 +89,211 @@ fn test_batch_raft_fallback() {
...
+            );
+        }
+
+        fn batch_raft(
+            &mut self,
+            ctx: RpcContext<'_>,
+            stream: RequestStream<BatchRaftMessage>,
+            sink: ClientStreamingSink<Done>,
+        ) {
+            // let status = RpcStatus::new(RpcStatusCode::Unimplemented, None);
```
Comment (hicqu) [inline tests/integrations/server/raft_client.rs]: Please remove these lines.

File: tests/integrations/server/raft_client.rs [hunk 4]
```diff
@@ -89,3 +89,211 @@ fn test_batch_raft_fallback() {
...
+
+    // Try to bind the mock server on a TCP port, and then do test.
+    for i in 0..100 {
+        let kv_service = MockKv(MockKvForRaft(Arc::clone(&counter)));
+        let port = 50000 + i;
+        let mut mock_server = match mock_kv_service(kv_service, "localhost", port) {
+            Ok(s) => s,
+            Err(_) => continue,
+        };
+        println!("port is {}",port);
```
Comment (hicqu) [inline tests/integrations/server/raft_client.rs]: Please remove the `println`.

File: tests/integrations/server/raft_client.rs [hunk 5]
```diff
@@ -89,3 +89,211 @@ fn test_batch_raft_fallback() {
...
+        mock_server.start();
+
+        let addr = format!("localhost:{}", port);
+
+        (0..50).for_each(|_| {
+            raft_client.send(1, &addr, RaftMessage::new()).unwrap();
+           // if let Err(err) = raft_client.send(1, &addr, RaftMessage::new()).unwrap() {
+           //     println!("{:?}",err), 
+           // };
+            thread::sleep(time::Duration::from_millis(10));
```
Comment (hicqu) [inline tests/integrations/server/raft_client.rs]: Don't need to sleep.

File: tests/integrations/server/raft_client.rs [hunk 6]
```diff
@@ -89,3 +89,211 @@ fn test_batch_raft_fallback() {
...
+            raft_client.flush();
+        });
+
+        println!("counter {:?}",counter.load(Ordering::SeqCst));
+
+       // thread::sleep(time::Duration::from_millis(1000));
+        println!("drop the server");
+        drop(mock_server);
+        (50..100).for_each(|_| {
+             raft_client.send(1, &addr, RaftMessage::new()).unwrap();
```
Comment (hicqu) [inline tests/integrations/server/raft_client.rs]: Can we check some conditions here?

File: tests/integrations/server/raft_client.rs [hunk 7]
```diff
@@ -89,3 +89,211 @@ fn test_batch_raft_fallback() {
...
+                       thread::sleep(time::Duration::from_millis(10));
+                       raft_client.flush();
+                       println!(" {} counter {}",n,counter.load(Ordering::SeqCst));
+             });
+             println!("counter {:?}",counter.load(Ordering::SeqCst));
+
+             assert!(counter.load(Ordering::SeqCst) > 50);
+             break;
+        }
+        assert!(counter.load(Ordering::SeqCst) >98);
```
Comment (hicqu) [inline tests/integrations/server/raft_client.rs]: Why `98`?

File: tests/integrations/server/raft_client.rs [hunk 8]
```diff
@@ -89,3 +89,211 @@ fn test_batch_raft_fallback() {
...
+        break;
+    }
+ pool.shutdown().wait().unwrap();
+}
+
+#[test]
+fn test_batch_raft_network_isolation() {
```
Comment (hicqu) [inline tests/integrations/server/raft_client.rs]: The test case needs `iptables`, and `iptables` needs root privillege. So please remove it, we can consider add it in schrodinger.

**Comment by zhangjinpeng1987** [comment]
Bad pr name.

**Pull request closed without merging**